todoapp/app/build.gradle

```diff
@@ -12,11 +12,20 @@ android {
         versionName "1.0"
 
         testInstrumentationRunner 'android.support.test.runner.AndroidJUnitRunner'
+
+        jackOptions {
+            enabled true
+        }
+    }
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
     }
 
     buildTypes {
         debug {
-            minifyEnabled true
+            minifyEnabled false
```

Comment:
Uhmm in that case comment and at least minify the release build type. http://stackoverflow.com/questions/37314760/android-how-to-perform-obfuscation-with-the-jack-compiler

It's important to compare APK size and method count.